util/migration/src/lib.rs

```diff
@@ -157,8 +160,12 @@ impl<T: SimpleMigration> Migration for T {
 		};
 
 		for (key, value) in iter {
-			if let Some((key, value)) = self.simple_migrate(key.into_vec(), value.into_vec()) {
-				batch.insert(key, value, dest)?;
+			if migration_needed {
+				if let Some((key, value)) = self.simple_migrate(key.into_vec(), value.into_vec()) {
+					batch.insert(key, value, dest)?;
+				}
+			} else {
+				batch.insert(key.into_vec(), value.into_vec(), dest)?;
```

Comment:
By default we are just altering the database, do we really need to overwrite the data?

Reply:
Unfortunately, we are not altering, but swapping :( 

https://github.com/paritytech/parity/blob/25b19835e316f53366df67b66244a8eee95bc865/util/migration/src/lib.rs#L291-L304

Reply:
yes, since it can fail we need to keep a backup until it completes